front/app/components/UI/RHFFileUploader/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from 'react';
+import FileUploader, { FileUploaderProps } from 'components/UI/FileUploader';
+
+import Error from 'components/UI/Error';
+import { Controller, useFormContext } from 'react-hook-form';
+import { UploadFile } from 'typings';
+import useRemoteFiles from 'hooks/useRemoteFiles';
+import { TResourceType } from 'resources/GetRemoteFiles';
+
+interface Props
+  extends Omit<
+    FileUploaderProps,
+    'onFileAdd' | 'onFileRemove' | 'files' | 'id'
+  > {
+  name: string;
+  resourceType: TResourceType;
+  resourceId: string | null;
+}
+
+const RHFFileUploader = ({
```

Comment:
Every input has a React hook form wrapper that handles the integration with React hook form.